crates/common/src/spin_env.rs

```diff
@@ -0,0 +1,19 @@
+//! Spin environment
+
+use std::path::PathBuf;
+
+/// Return a path to the `spin` binary; the first of these that is available:
+/// - the `SPIN_BIN_PATH` environment variable
+/// - the current executable ([`std::env::current_exe`])
+/// - the constant `"spin"` (resolved by e.g. `$PATH`)
+pub fn spin_bin_path() -> PathBuf {
+    if let Some(path) = std::env::var_os("SPIN_BIN_PATH") {
```

Comment:
This environment variable will be relevant to plugins to "get back to" the Spin binary that launched them, but should probably be ignored within Spin itself.  There's bountiful potential for confusion and hard-to-debug inconsistencies if `spin doctor` or `spin up` were to run a _different version_ of Spin without the user realising it.

Reply:
Hmm...I was specifically thinking of the case where a plugin might want to shell out to `spin doctor` here. Someone accidentally setting `SPIN_BIN_PATH` to something surprising seems like a pretty unlikely scenario to me.

Reply:
Possibly silly idea but we could also just print out a warning if `SPIN_BIN_PATH` is ever set when just directly running `spin`. That would prevent this edge case.

Reply:
But in the case where a plugin is shelling out to `spin doctor`, we are already inside a Spin process by the time this function gets called, so `current_exe` will report the right thing, won't it?

(Also, at the risk of derailing the serious question above, here's an "Edgiest of the Edge" case: Consider a plugin that intentionally shells out to `spin doctor` or `spin up` using a copy of Spin that was _not_ the one that called it for uh reasons.  Now suppose that `spin doctor` calls `spin build`. or `spin up` calls `spin trigger`.  With this override in place, those double-shells are going to call to the _original_ copy of Spin, not the copy that the plugin asked for.)

Reply:
> But in the case where a plugin is shelling out to spin doctor, we are already inside a Spin process by the time this function gets called, so current_exe will report the right thing, won't it?

Sorry, I wrote that wrong; I'm thinking of a plugin using the `spin-doctor` crate directly, then a spin doctor treatment shelling out to e.g. `spin build`.

Reply:
In that case maybe have the `spin-doctor` crate export a function explicitly for plugins, which could simply fail if (somehow) `SPIN_BIN_PATH` were not set rather than falling back to `current_exe` - then the plugin would need to link only to `spin-doctor` rather than also bringing in `spin-common` (which is not really what we want to present as The Face Of Spin).

I think what I'm trying to get at (admittedly not very clearly) is that plugins and Spin itself have entirely different strategies for locating the Spin binary, and there is no need or benefit for one function to try to serve both those purposes.

Reply:
Thinking about this and manifest resolution and no doubt other things, seems like a "plugin utils" library (to help plugins behave consistently with Spin) might be in our futures, though given that plugins are intentionally not bound to Rust, it could end up becoming a labour to ship across multiple languages.

Reply:
Removed.